sdk/core/azure-core/src/http/response.cpp

```diff
@@ -17,4 +18,38 @@ std::map<std::string, std::string> const& Response::getHeaders() { return this->
 
 BodyStream* Response::getBodyStream() { return m_bodyStream; }
 
-BodyBuffer* Response::getBodyBuffer() { return m_bodyBuffer; }
+std::vector<uint8_t>& Response::getBodyBuffer() { return m_bodyBuffer; }
+
+std::string Response::getStringBody()
+{
+  return std::string(m_bodyBuffer.begin(), m_bodyBuffer.end());
+}
+
+std::string Response::getHttpVersion() { return m_httpVersion; }
+
+namespace
+{
+inline bool is_string_equals_ignore_case(std::string const& a, std::string const& b)
+{
+  return a.length() == b.length()
```

Comment:
The perf of this will be awful since it invokes a callback function once per character!
You should write the loop yourself and compare characters.